src/main/java/module-info.java

```diff
@@ -0,0 +1,7 @@
+module HQDM.model.main {
```

Comment:
I am no longer able to generate javadocs for this project. 

running `mvn javadog:aggregate` produces the following error: 
```
org.apache.maven.reporting.MavenReportException: 
Exit code: 1 - Loading source file .../HQDM/src/main/java/module-info.java...
error: module not found: HQDM.model.main 
```

Reply:
I was able to fix this locally by converting the maven build to use a `hqdm` submodule and a parent `pom.xml` and moving the `src` folder into a subdirectory `hqdm`. This is likely to make merging the subsequent PRs very difficult so I propose doing this after those PRs are merged and accepting the javadoc issue until then. 

What do you think?

Reply:
I also changed the module name to `uk.gov.gchq.hqdm` to match the recommended naming conventions for modules, and will do something similar on the other repos I'm working on as well.

Reply:
I'd prefer to do this change now to avoid contaminating the commit history with bad code. We'll need to rebase the remaining PRs onto this one to avoid conflicts with these changes anyway so it might as well be a part of this PR.

Reply:
Ok done - hopefully Git will be able to keep track of everything during the rebase.